tensorflow_addons/losses/pinball.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements pinball loss."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow.keras.backend as K
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+@tf.function
+def pinball(y_true, y_pred, tau):
+  """Computes the pinball loss between `y_true` and `y_pred`.
+  Usage:
+  ```python
+  loss = pinball([0., 0., 1., 1.], [1., 1., 1., 0.], tau=.1)
+  print('Loss: ', loss.numpy())  # Loss:
+  ```
+  Args:
+    tau: a float between 0 and 1 the slope of the pinball loss. In the context
+    of quantile regression, the value of alpha determine the conditional
+    quantile level.
+  """
+    delta_y = y_true - y_pred
+    return K.maximum(tau * delta_y, (tau - 1) * delta_y)
+
+
+@keras_utils.register_keras_custom_object
+class Pinball(tf.keras.losses.LossFunctionWrapper):
```

Comment:
Probably inherit from `tf.keras.losses.Loss` because `LossFunctionWrapper` is not a public API.